File: Competitive Programming 3/Section 1.4/Games (Others), Easier/489 - Hangman Judge/Python/main.py
def main():
    while True:
        roundd = int(input())
        if roundd == -1:
            break
        objective,visited,wrong=set(),set(),set()
        print('Round '+str(roundd))
        for c in input():
            objective.add(c)
        for c in input():
            if c in objective:
                visited.add(c)
                if len(visited)==len(objective):
                    break
            else:
                wrong.add(c)
                if len(wrong)>6:
                    #loses
                    break
        print('You '+('lose' if len(wrong)>6 else ('win' if len(visited)==len(objective) else 'chickened out'))+'.')
main()

File: Competitive Programming 3/Section 1.4/Games (Others), Easier/489 - Hangman Judge/C++/main.cpp
#include<bits/stdc++.h>

using namespace std;

int roundd;
string line;

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    while(cin>>roundd,roundd!=-1){
        set<int> objective,visited,wrong;
        cout << "Round " << roundd << "\n";
        cin >> line;
        for(char c:line){
            objective.insert(c);
        }
        cin >> line;
        for(char c:line)
            if(objective.find(c)!=objective.end()){
                visited.insert(c);
                if(visited.size()==objective.size())
                    break;
            }else{
                wrong.insert(c);
                if(wrong.size()>6)//loses
                    break;
            }
        cout << "You " << (wrong.size()>6 ? "lose" : (visited.size()==objective.size() ? "win" : "chickened out")) << ".\n";
    }
    return 0;
}